common/src/main/java/org/jboss/hal/testsuite/creaper/ResourceVerifier.java

```diff
@@ -0,0 +1,113 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.hal.testsuite.creaper;
+
+import java.io.IOException;
+
+import org.junit.Assert;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+
+/**
+ * Created by pjelinek on Nov 16, 2015
+ */
+public class ResourceVerifier {
+
+    private Address resourceAddress;
+    private Operations ops;
+
+    public ResourceVerifier(OnlineManagementClient client) {
+        this.ops = new Operations(client);
+    }
+
+    public ResourceVerifier(Address resourceAddress, OnlineManagementClient client) {
+        this.resourceAddress = resourceAddress;
+        this.ops = new Operations(client);
+    }
+
+    /**
+     * Resource address setter.
+     */
+    public ResourceVerifier forResource(Address address) {
+        this.resourceAddress = address;
+        return this;
+    }
+
+    /**
+     * Verifies if previously set resource exists in model.
+     */
+    public ResourceVerifier verifyResource(boolean expected) throws IOException, OperationException {
+        addressIsSetCheck();
+        boolean resourceExists = ops.exists(resourceAddress);
+        if (expected) {
+            Assert.assertTrue("Resource '" + resourceAddress + "' should exist!", resourceExists);
+        } else {
+            Assert.assertFalse("Resource '" + resourceAddress + "' should NOT exist!", resourceExists);
+        }
+        return this;
+    }
+
+    /**
+     * Verifies the value of attribute in model.
+     */
+    public ResourceVerifier verifyAttribute(String attributeName, String expectedValue) throws IOException {
+        addressIsSetCheck();
+        ModelNodeResult actualResult = ops.readAttribute(resourceAddress, attributeName);
+        actualResult.assertDefinedValue();
+        Assert.assertEquals("Attribute value is different in model!", expectedValue, actualResult.stringValue());
+        return this;
+    }
+
+    /**
+     * @throws AssertionError if previously provided resource exists and has attribute with attributeName
+     * with value equal to notExpectedValue.
+     */
+    public ResourceVerifier verifyAttributeNotEqual(String attributeName, String notExpectedValue) throws IOException {
+        addressIsSetCheck();
+        ModelNodeResult actualResult = ops.readAttribute(resourceAddress, attributeName);
+        if (actualResult.isSuccess() && actualResult.hasDefinedValue() && actualResult.stringValue().equals(notExpectedValue)) {
+            throw new AssertionError(attributeName + " should not have value " + notExpectedValue);
+        }
+        return this;
+    }
```

Comment:
Not all resources in model are Strings, wouldn't it make sense to add methods for List, Boolean, .... or make the comparison compare ModelNodes?